File: algorithms/1_easy/I/00226_invert_binary_tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invert_tree_helper(self, curr_node):

        if curr_node == None:
            return

        temp = curr_node.left
        curr_node.left = curr_node.right
        curr_node.right = temp

        self.invert_tree_helper(curr_node.left)
        self.invert_tree_helper(curr_node.right)

    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.invert_tree_helper(root)

        return root

File: algorithms/1_easy/I/00226_invert_binary_tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    void invert_tree_helper(TreeNode* curr_node) {

        if (curr_node == nullptr) {
            return;
        }

        std::swap(curr_node->left, curr_node->right);

        invert_tree_helper(curr_node->left);
        invert_tree_helper(curr_node->right);
    }
public:
    TreeNode* invertTree(TreeNode* root) {

        invert_tree_helper(root);

        return root;
    }
};
